playbooks/adjust-cinder-backend.yml

```diff
@@ -0,0 +1,239 @@
+    service:
+      name: "{{ item[1] }}"
+      state: restarted
+      must_exist: false
+    run_once: True
+    delegate_to: "{{ item[0] }}"
+    when: set_cinder_host or update_cinder_backend
+    with_nested:
+      - "{{ play_hosts }}"
+      - ['cinder-api', 'cinder-scheduler', 'cinder-volume']
+
+  - name: restart cinder backup service
+    service:
+      name: cinder-backup
+      state: restarted
+      must_exist: false
+    run_once: True
+    delegate_to: "{{ item }}"
+    when:
+      - swift.enabled|default('False')|bool
+      - (set_cinder_host or update_cinder_backend)
+    with_items:
+      - "{{ play_hosts }}"
+
+  tasks:
+  - name: gather facts for adjusting cinder service host
+    shell: grep '^host \?= \?ceph' /etc/cinder/cinder.conf
+    ignore_errors: true
+    changed_when: false
+    register: check_cinder_host
+
+  - name: gather facts for adjusting cinder backend name
+    shell: grep '^enabled_backends \?= \?rbd_volumes' /etc/cinder/cinder.conf
+    ignore_errors: true
+    changed_when: false
+    register: check_cinder_backend
+
+  - name: gather facts for existing default_volume_type
+    shell: grep '^default_volume_type' /etc/cinder/cinder.conf
+    ignore_errors: true
+    changed_when: false
+    register: check_default_volume_type
+
+  - name: set fact for set_cinder_host
+    set_fact:
+      set_cinder_host: True
+    when: check_cinder_host.rc != 0
+
+  - name: set fact for update_cinder_backend
+    set_fact:
+      update_cinder_backend: True
+    when: check_cinder_backend.rc == 0
+
+  - name: set fact for set_default_volume_type
+    set_fact:
+      set_default_volume_type: True
+    when: check_default_volume_type.rc != 0
+
+  - name: set fact for hybrid cinder backend
+    set_fact:
+      cinder_backend_name: rbd_hybrid
+      cinder_volume_type: CEPH_HYBRID
+    when: ceph.bcache_ssd_device is defined
+
+  - name: set fact for ssd cinder backend
+    set_fact:
+      cinder_backend_name: rbd_ssd
+      cinder_volume_type: CEPH_SSD
+    when: ceph.bcache_ssd_device is not defined
+
+  - name: stop all cinder services
+    command: echo "stopping all cinder services"
+    when: set_cinder_host or update_cinder_backend
+    notify:
+      - stop cinder services
+      - stop cinder backup service
+
+  - name: trigger stop all cinder services
+    meta: flush_handlers
+
+  - name: clear cinder services from db
+    command: mysql -e "DELETE cinder.services FROM cinder.services;"
+    when: set_cinder_host or update_cinder_backend
+    run_once: true
+    notify:
+      - restart cinder services
+      - restart cinder backup service
+
+  - name: update cinder volumes host entries in db
+    command: mysql -e "
+             DELETE cinder.services FROM cinder.services;
+             UPDATE cinder.volumes SET host='ceph@{{ cinder_backend_name }}' WHERE host is not NULL;
+             UPDATE cinder.backups SET host='ceph' WHERE host is not NULL;"
+    when: set_cinder_host or update_cinder_backend
```

Comment:
This task and the previous one seem to be doing much of the same thing. Should it just be one command?